### PR TITLE
Replace Geometry_cff with GeometryDB_cff in RecoJets

### PR DESCRIPTION
**Pull request by jeongeun**
**PR description:**
Review on the Reco part of https://github.com/cms-sw/cmssw/issues/31113

`process.load("Configuration.StandardSequences.Geometry_cff")`
 was outdated https://github.com/cms-sw/cmssw/pull/8810 
It should be replaced with
`process.load("Configuration.StandardSequences.GeometryDB_cff")`

In this PR, RecoJets configuration files (4 files) are fixed.
```
modified:   RecoJets/JetAnalyzers/test/JetIdExample.py
modified:   RecoJets/JetAssociationProducers/test/pileupJetAnalysis_cfg.py
modified:   RecoJets/JetAssociationProducers/test/test_extrapolation_cff.py
modified:   RecoJets/JetProducers/test/runSubjetFilter_cfg.py
```

#### PR validation:
Tested in CMSSW_12_5_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: RecoJets/JetAnalyzers/test/JetIdExample.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
    
 process = cms.Process("Ana")
 process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_cff')
-process.load("Configuration.StandardSequences.Geometry_cff")
+process.load("Configuration.StandardSequences.GeometryDB_cff")
 process.GlobalTag.globaltag = GLOBAL_TAG
```
Comment (clacaputo) [inline RecoJets/JetAnalyzers/test/JetIdExample.py:46]: Hi @jeongeun `GLOBAL_TAG` is initialized with old GT; moreover, the input files used are from 2010.
Did you try to run it? Maybe we can directly remove this?

@cms-sw/jetmet-pog-l2 , what do you think?

Comment (clacaputo) [reply RecoJets/JetAnalyzers/test/JetIdExample.py:46]: Hi @cms-sw/jetmet-pog-l2 , any thoughts about it?

**Comment by jpata** [comment]
kind ping @cms-sw/pf-l2 @laurenhay

**Comment by kdlong** [comment]
@jpata is this for us or for JME?

**Comment by jpata** [comment]
sorry it was for @cms-sw/jetmet-pog-l2 (Lauren is on double duty and I got confused).

**Comment by laurenhay** [comment]
Thanks for the ping! From what I can see I think for sure RecoJets/JetAnalyzers/test/JetIdExample.py is deprecated since it relies on processors that were removed in PR https://github.com/cms-sw/cmssw/pull/19406.
Likely we can remove it or should at least update it. Waiting for feedback on the other files.

**Comment by clacaputo** [comment]
> @joosep Thanks for the ping! From what I can see I think for sure RecoJets/JetAnalyzers/test/JetIdExample.py is not in use since it loads processors that were removed in PR #19406. Likely we can remove it or should at least update it. Waiting for feedback on the other files.

Hi @laurenhay , any update on the other files? Thanks

**Comment by clacaputo** [comment]
A kindly ping to @laurenhay

**Comment by laurenhay** [comment]
Hey @clacaputo sorry for the delay I think the consensus is they can go.
I asked the current JME conveners (a couple times) and there were no complaints at deleting or requests to keep them.
Also asked @rappoccio and he said they can go.

**Comment by clacaputo** [comment]
> Hey @clacaputo sorry for the delay I think the consensus is they can go. I asked the current JME conveners (a couple times) and there were no complaints at deleting or requests to keep them. Also asked @rappoccio and he said they can go.

Thanks for checking. @jeongeun could you please remove them?

**Comment by alkaloge** [comment]
sorry for replying late - JME has no objections

**Comment by jeongeun** [comment]
@cmsbuild please test

**Comment by mandrenguyen** [comment]
please test

**Comment by clacaputo** [comment]
There are some differences in JetMET folder of DMQ bin-by-bin comparison for wf `20900` [1] (and also other wfs) that shouldn't be there given the PR's nature. Any idea @cms-sw/jetmet-pog-l2 ?

[1] https://tinyurl.com/2c82kenn

**Comment by clacaputo** [comment]
> There are some differences in JetMET folder of DMQ bin-by-bin comparison for wf `20900` [1] (and also other wfs) that shouldn't be there given the PR's nature. Any idea @cms-sw/jetmet-pog-l2 ?
> 
> [1] https://tinyurl.com/2c82kenn

Update: it seems to be a widespread issue

**Comment by clacaputo** [comment]
+reconstuction

- technical PR: removing some unused code
- changes observed are not related to the PR

**Comment by perrotta** [comment]
+1

**Pull request merged**